src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -79,6 +80,18 @@ public static Parameter FromInputParameter(in InputParameter operationParameter,
                 RequestLocation: requestLocation);
         }
 
+        private static Constant? GetDefaultValue(InputParameter operationParameter, TypeFactory typeFactory) => operationParameter switch
+        {
+            { DefaultValue: not null } => BuilderHelpers.ParseConstant(operationParameter.DefaultValue.Value, typeFactory.CreateType(operationParameter.DefaultValue.Type)),
+            { NameInRequest: var nameInRequest } when nameInRequest.Equals(RequestHeader.RepeatabilityRequestId, StringComparison.OrdinalIgnoreCase) =>
```

Comment:
If the Repeatability parameter has a default value, it will not enter this option. Maybe we need to put repeatability parameter at the top.

Reply:
> If the Repeatability parameter has a default value, it will not enter this option. Maybe we need to put repeatability parameter at the top.

I think that's a wrong usage. People should not set a default value for repeatability headers. The value **must be different** per call. Think of the following scenario:
- The first time we call `Foo()`. `Repeatability-Request-ID` is a default value (`abc`).
- The server side will remember `abc`.
- The next time we call `Foo()` with the same header value. Server will treat it as a retry, and will not do anything.

Reply:
Yes. It is rare to set default value for repeatability. But according to the code, it may have this problem.

Reply:
> Yes. It is rare to set default value for repeatability. But according to the code, it may have this problem.

If we set the default value, the repeatability request feature will not work. See [spec 3.1.1 Repeatability-Request-ID](https://docs.oasis-open.org/odata/repeatable-requests/v1.0/cs01/repeatable-requests-v1.0-cs01.html#sec_RepeatabilityRequestID):
> The value of the Repeatability-Request-ID is an opaque string representing a client-generated, globally unique for all time, identifier for the request.

The value must be **globally unique** to mark every request. We should not set the same value for different requests.

Reply:
What I mean is that to consider the function itself (ignoring the backend limitation of the repeatability headers ), the first option `{ DefaultValue: not null }` may meet first and then following options to handle repeatability headers may not executed.
So it may be better to put special options head of the more general one.

Reply:
@archerzz can you open an issue in typespec repo to add a linter rule for never setting a default value for these headers?

Reply:
I tend to treat this as an error in original spec. Better to generate the wrong codes so that the spec owner can find the problem. Another way to prevent it is to add a linter rule.